src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/IconTintColor/IconTintColorBehavior.ios.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.ComponentModel;
+using Microsoft.Maui.Platform;
+using UIKit;
+
+namespace CommunityToolkit.Maui.Behaviors;
+public partial class IconTintColorBehavior
+{
+	/// <inheritdoc/>
+	protected override void OnAttachedTo(View bindable, UIView platformView) =>
+		ApplyTintColor(platformView, bindable, TintColor);
+
+	/// <inheritdoc/>
+	protected override void OnDetachedFrom(View bindable, UIView platformView) =>
+		ClearTintColor(platformView, bindable);
+
+	void ClearTintColor(UIView platformView, View element)
+	{
+		switch (platformView)
+		{
+			case UIImageView imageView:
+				element.PropertyChanged -= OnImageViewTintColorPropertyChanged;
+				if (imageView.Image is not null)
+				{
+					imageView.Image = imageView.Image.ImageWithRenderingMode(UIImageRenderingMode.AlwaysOriginal);
+				}
+
+				break;
+			case UIButton button:
+				element.PropertyChanged -= OnButtonTintColorPropertyChanged;
+				if (button.ImageView?.Image is not null)
+				{
+					var originalImage = button.CurrentImage.ImageWithRenderingMode(UIImageRenderingMode.AlwaysOriginal);
+					button.SetImage(originalImage, UIControlState.Normal);
+				}
+
+				break;
+		}
+	}
+
+	void ApplyTintColor(UIView platformView, View element, Color? color)
+	{
+		if (color is null)
+		{
+			ClearTintColor(platformView, element);
+			return;
+		}
+
+		switch (platformView)
+		{
+			case UIImageView imageView:
+				SetUIImageViewTintColor(imageView, element, color);
+				break;
+			case UIButton button:
+				SetUIButtonTintColor(button, element, color);
+				break;
+		}
+	}
+
+	void SetUIButtonTintColor(UIButton button, View element, Color color)
+	{
+		if (button.ImageView.Image is null)
+		{
+			element.PropertyChanged += OnButtonTintColorPropertyChanged;
+		}
+		else
+		{
+			var templatedImage = button.CurrentImage.ImageWithRenderingMode(UIImageRenderingMode.AlwaysTemplate);
```

Comment:
Just curious - how come we use `UIImageRenderingMode.AlwaysTemplate` here, but use `UIImageRenderingMode.AlwaysOriginal` in `ClearTintColor()`?

Reply:
Setting rendering mode to `AlwaysTemplate` tells the system that we want to use it as an image mask which we can apply any color we want.